### PR TITLE
refactor(events): use of classes for scalability & maintainability

### PR DESCRIPTION
**Pull request by jacoobes**
This breaks up the long spaghetti code that was handling events into classes that are more maintainable and scalable

### CODE REVIEW

**Comment by EvolutionX-10** [comment]
is this pr ready for review?

**Comment by jacoobes** [comment]
Yes

File: src/handler/events/eventsHandler.ts
Comment (EvolutionX-10) [inline src/handler/events/eventsHandler.ts:3]: You prefer this (1. current) or
2.
```ts
import { Subject, type Observable } from 'rxjs';
```
or
3.
```ts
import { Observable, Subject } from 'rxjs';
```

File: src/handler/events/messageHandler.ts
Comment (EvolutionX-10) [inline src/handler/events/messageHandler.ts:48]: > 1.
```ts
if (!this.wrapper.defaultPrefix) return;
```
> 2.
```ts
if (this.wrapper.defaultPrefix === undefined) return;
```
> 3.
```ts
if (typeof this.wrapper.defaultPrefix === 'undefined') return;
```

File: src/handler/events/observableHandling.ts
Comment (EvolutionX-10) [inline src/handler/events/observableHandling.ts:106]: why need to cast it? what does it infer if not casted?

File: src/handler/events/readyHandler.ts
Comment (EvolutionX-10) [inline src/handler/events/readyHandler.ts:52]: how bout building another enum for all sern events?
enums are style of v14 so 👀

**Review by EvolutionX-10 — CHANGES_REQUESTED** [review]
GG
Make sure to test it locally on a bot before merging!
As well as run prettier

File: src/handler/events/eventsHandler.ts
Comment (jacoobes) [reply src/handler/events/eventsHandler.ts:3]: Definitely two, i didnt know u could do that

File: src/handler/events/messageHandler.ts
Comment (jacoobes) [reply src/handler/events/messageHandler.ts:48]: 2

File: src/handler/events/observableHandling.ts
Comment (jacoobes) [reply src/handler/events/observableHandling.ts:106]: it is inferred as unknown, but good point. i dont think casting is needed here

File: src/handler/events/readyHandler.ts
Comment (jacoobes) [reply src/handler/events/readyHandler.ts:52]: we will see. You can register events by file name, so not really sure how enums would play into this

File: src/handler/events/eventsHandler.ts
Comment (EvolutionX-10) [reply src/handler/events/eventsHandler.ts:3]: xD

File: src/handler/events/readyHandler.ts
Comment (EvolutionX-10) [reply src/handler/events/readyHandler.ts:52]:
```ts
enum Events {
	ModuleRegister = 'module.register'
	// etc
}
```

Comment (EvolutionX-10) [reply src/handler/events/readyHandler.ts:52]: also wait why would file name be responsible for events
i don't need to have the event name same as file name, so i should be asked for an event name
now we can combine djs event names and sern events name and export it as `Events` so users can use it

File: src/handler/events/observableHandling.ts
Comment (jacoobes) [reply src/handler/events/observableHandling.ts:106]: nvm, its needed 
image

**Comment by jacoobes** [comment]
Should i also just remove cz as we are using release please?

**Comment by EvolutionX-10** [comment]
> Should i also just remove cz as we are using release please?

sure, also cz has no relation with release please lol, just make sure to follow conventional commits properly
the initial pr name was incorrect

**Comment by EvolutionX-10** [comment]
don't remove it in this pr, either directly commit in main or create new pr for that

File: src/handler/events/observableHandling.ts
Comment (EvolutionX-10) [reply src/handler/events/observableHandling.ts:103]: alright

**Comment by jacoobes** [comment]
and what about standard version as well?

**Comment by EvolutionX-10** [comment]
yup that can be removed, i removed it from cli as well, it's soft deprecated in favor of release please according to the devs of standard version itself

**Comment by jacoobes** [comment]
done

**Comment by jacoobes** [comment]
check the clean up imports pr and then ill test it one more time with cheemsBanker

**Comment by jacoobes** [comment]
hmmm
getting some issues with dev version of discord.js, specifically
image
discord.js shapeshift is causing issues. will investigate further

**Comment by jacoobes** [comment]
Might be a local issue and not discordjs. Will wait a few hours, ive had trouble with my pc and files in general due to onedrive

**Comment by EvolutionX-10** [comment]
oh my god these errors are so detailed

**Comment by jacoobes** [comment]
Yeah no luck. Won't push this branch until this error is resolved

**Comment by jacoobes** [comment]
fixed, i dont know what hapepned but restarting computer did the trick last night

**Comment by EvolutionX-10** [comment]
oh

**Review by Murtatrxx — APPROVED** [review]
Seems like good improvements has been made 👍

**Comment by jacoobes** [comment]
Great, sounds good
Will wait for evo'a approval cuz he is git master

**Comment by Murtatrxx** [comment]
wym by "git master"

**Comment by jacoobes** [comment]
knows what to do for git prs, like force pushing , rebasing, etc

File: src/handler/events/interactionHandler.ts
Comment (EvolutionX-10) [inline src/handler/events/interactionHandler.ts:35]: Empty????

**Comment by Murtatrxx** [comment]
Oh ok

File: src/handler/events/interactionHandler.ts
Comment (Murtatrxx) [reply src/handler/events/interactionHandler.ts:35]: I think it's a todo

Comment (jacoobes) [reply src/handler/events/interactionHandler.ts:35]: strange, its on my local branch

Comment (EvolutionX-10) [reply src/handler/events/interactionHandler.ts:35]: might update and sync it

Comment (jacoobes) [reply src/handler/events/interactionHandler.ts:35]: oh, let me remove this.

Comment (jacoobes) [reply src/handler/events/interactionHandler.ts:35]: would i merge or rebase this ?

Comment (EvolutionX-10) [reply src/handler/events/interactionHandler.ts:35]:
```bash
git commit # if you haven't yet
git pull --rebase
git push --force
```
that should do most likely

**Comment by jacoobes** [comment]
uh i did that idk if that was correct

**Review by EvolutionX-10 — APPROVED** [review]
Cheers

**Comment by EvolutionX-10** [comment]
Was a merge and not a rebase but can't help without discord properly
anyways I squashed it so not a big deal

**Comment by jacoobes** [comment]
Ah okay, glad it worked out though

**Comment by jacoobes** [comment]
Oh yeah, should i bump version on main?

**Comment by Murtatrxx** [comment]
V1.0.0?

**Comment by jacoobes** [comment]
this is only a refactor branch so i was thinking v1.2.0-beta

**Comment by Murtatrxx** [comment]
Then I'm preparing the release message

**Comment by jacoobes** [comment]
nvm, semver states this should be a patch version

    MAJOR version when you make incompatible API changes,
    MINOR version when you add functionality in a backwards compatible manner, and
    PATCH version when you make backwards compatible bug fixes.

**Comment by jacoobes** [comment]
so, 1.1.1 👍

**Comment by Murtatrxx** [comment]
👍

**Comment by Murtatrxx** [comment]
Not posting v0.0.x releases

**Comment by jacoobes** [comment]
Ok sounds good

**Comment by jacoobes** [comment]
I will just say 1.1.1 beta got released. and link new npm link

**Comment by jacoobes** [comment]
good work everyone

**Comment by Murtatrxx** [comment]
Wait I can do it, you want it to anyways

**Pull request merged**
